internal/datanode/data_node.go

```diff
@@ -901,6 +900,50 @@ func (node *DataNode) Import(ctx context.Context, req *datapb.ImportTaskRequest)
 	return resp, nil
 }
 
+// AddSegment adds the segment to the current DataNode.
+func (node *DataNode) AddSegment(ctx context.Context, req *datapb.AddSegmentRequest) (*commonpb.Status, error) {
+	log.Info("adding segment to DataNode flow graph",
+		zap.Int64("segment ID", req.GetSegmentId()),
+		zap.Int64("collection ID", req.GetCollectionId()),
+		zap.Int64("partition ID", req.GetPartitionId()),
+		zap.String("channel name", req.GetChannelName()),
+		zap.Int64("# of rows", req.GetRowNum()))
```

Comment:
Why is "# of rows"?

Reply:
number of rows in the segment, we need this to update update segment stat in: `ds.replica.updateStatistics(req.GetSegmentId(), req.GetRowNum())`